systemtest/gencorpora/catbulk.go

```diff
@@ -53,44 +62,93 @@ func NewCatBulkServer() (*CatBulkServer, error) {
 		return nil, err
 	}
 
-	writer, err := getWriter(gencorporaConfig.WritePath)
+	writer, err := os.Create(gencorporaConfig.CorporaPath)
 	if err != nil {
 		return nil, err
 	}
 
 	addr := listener.Addr().String()
+	metaUpdateChan := make(chan docsStat)
 	return &CatBulkServer{
 		listener: listener,
 		Addr:     addr,
 		server: &http.Server{
 			Addr:    addr,
-			Handler: handleReq(writer),
+			Handler: handleReq(metaUpdateChan, writer),
 		},
-		writer: writer,
+		writer:         writer,
+		metaUpdateChan: metaUpdateChan,
+		metaWriteDone:  make(chan struct{}),
 	}, nil
 }
 
 // Serve starts the fake ES server on a listener.
 func (s *CatBulkServer) Serve() error {
+	go func() {
+		if err := s.metaWriter(); err != nil {
+			log.Println("failed to write metadata", err)
+		}
+	}()
+
 	if err := s.server.Serve(s.listener); err != http.ErrServerClosed {
```

Comment:
```suggestion
	if err := s.server.Serve(s.listener); err != nil && err != http.ErrServerClosed {
```

**nit**: Non blocking, but I think this should check for `nil` before returning. It doesn't change the behaviour of the code but it helps with understanding the flow

Reply:
Good catch, thanks!